src/ingest/TimeRow.h

```diff
@@ -142,8 +142,7 @@ class TimeRow : public nebula::surface::RowData {
       break;
     }
     case nebula::meta::TimeType::MACRO: {
-      // TODO(cao) - support only one macro for now, need to generalize it
-      if (ts.pattern == "date") {
+      if (nebula::meta::extractPattern(ts.pattern) != nebula::meta::PatternMacro::INVALID) {
         return [mdate](const nebula::surface::RowData*) {
```

Comment:
nit: this could be renamed too

Reply:
I meant mdate - could you rename it to watermark as well?

Reply:
fixed